.github/workflows/build.yml

```diff
@@ -1,11 +1,11 @@
-name: End-to-end Tests
+name: Lint, Build and Test
```

Comment:
Validating the project and running e2e tests on the codebase are two different concerns in the CI. I suggest leaving this workflow as is and creating a second `validate.yml` workflow similar to this one whose work is to "Validate" the project. (see my comment below on why I propose the name validate)